providers/bulksms/src/lib/bulksms.provider.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  ISmsOptions,
+  ISmsProvider,
+} from '@novu/stateless';
+
+if (!globalThis.fetch) {
+  // eslint-disable-next-line global-require
+  globalThis.fetch = require('node-fetch');
+}
+
+export class BulkSmsSmsProvider implements ISmsProvider {
+  id = 'bulksms';
+  channelType = ChannelTypeEnum.SMS as ChannelTypeEnum.SMS;
+  private BASE_URL = 'http://login.bulksmsoffers.com/api/sendhttp.php?';
+
+  constructor(
+    private config: {
+      authKey?: string;
+      sender?: string;
+      route?: string;
+      dltEntity?: string;
+    }
+  ) {}
+
+  private parseHeaderDate(headers: Headers): string {
+    const date = headers.get('Date');
+
+    return date ? new Date(date).toISOString() : new Date().toISOString();
+  }
+
+  async sendMessage(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const baseMessage = {
+      authKey: this.config.authKey,
+      to: options.to,
+      sender: this.config.sender,
+      route: this.config.route,
+      message: encodeURIComponent(options.content),
+    };
+
+    const url =
+      this.BASE_URL +
+      'authkey=' +
+      baseMessage.authKey +
+      '&mobiles=' +
+      baseMessage.to +
+      '&message=' +
+      baseMessage.message +
```

Comment:
Should we do encodeUriComponent component for those fields here?

Reply:
> Should we do encodeUriComponent component for those fields here?

@scopsy we only need to encodeUriComponent the message part only in line no. 41 (i.e the content part) rest we don't need to do